ccmbioinfo/lfs
# CARDS-1623: Add basic support for DICOM image storage

This PR implements CARDS-1623 by making the `cards:DicomAnswer` data type available which allows for the creation of _Questionnaires_ containing DICOM based answers. The _DICOM Answer_ is based on the _File Answer_ with the extension of that when a DICOM file is uploaded, a preview is displayed in the Questionnaire and the DICOM file's metadata is made available in the _Notes_ section for that DICOM question. The original DICOM file is also available for download through the completed Form.

To test:

1. Build this `CARDS-1623` branch (`mvn clean install`).
2. Start CARDS in _ClinDig_ mode (`./start_cards.sh --project cards4clindig --dev`).
3. Point your browser to `http://localhost:8080` and login as `admin`:`admin`.
4. Create a new _Questionnaire_ of type _Imaging_ along with a new associated _Subject_.
5. Download the file `MR-MONO2-16-knee.gz` from https://barre.dev/medical/samples/#MR-MONO2-16-knee and extract it to a DICOM image with the command `cat MR-MONO2-16-knee.gz | gzip -d > MR-MONO2-16-knee.dcm`.
6. Upload the file `MR-MONO2-16-knee.dcm` as an answer to the DICOM question in CARDS.
7. Verify that a preview is displayed and that the associated metadata is correctly populated in that question's _Notes_ field.
8. Try editing the _Notes_ and re-visiting the _Form_. All should work as expected.
9. You may also wish to test with some of the other DICOM files available on https://barre.dev/medical/samples/
